Assets/Editor/CreatureHydrator.cs

```diff
@@ -96,14 +106,56 @@ private static void HydrateCreature(UnitDefinition unit, string generaName, stri
             // Slice the sprite sheet
             Sprite[] sprites = SliceSpriteSheet(texture, animationSpec);
 
-            if (animation.name.Equals("Idle", StringComparison.OrdinalIgnoreCase) && sprites.Length > 0)
+            if (sprites == null || sprites.Length == 0)
             {
-                idleSprite = sprites[0];
+                Debug.LogError($"No sprites found for animation {animation.name} in unit {unitName}");
+                continue;
             }
 
             // Create and add AnimationClip to AnimatorController
             AnimationClip clip = CreateAnimationClip(sprites, animationSpec.frames, animation.name, animationSpec.frameRate, animationFolderPath);
-            animatorController.AddMotion(clip);
+            AnimatorState state = animatorController.layers[0].stateMachine.AddState(animation.name);
+
+            if (animation.name.Equals("Idle", StringComparison.OrdinalIgnoreCase))
```

Comment:
I don't think this will NPE or C# equivalent, but wonder if it should be `"Idle".equals(animation.name, StringComparison.OrdinalIgnoreCase)` for all of these

Or honestly with 3 conditions, maybe a switch? https://www.w3schools.com/cs/cs_switch.php

Reply:
Then just use lower case https://stackoverflow.com/a/2334141/10061484 and compare instead of ordinalignorecase?

Reply:
Added an enum for BaseAnimation